scipy/stats/tests/test_mstats_basic.py

```diff
@@ -1381,19 +1381,6 @@ def test_vs_nonmasked(self):
         res2 = mstats.ttest_1samp(outcome[:, 0], 1)
         assert_allclose(res1, res2)
 
-        # 2-D inputs
-        res1 = stats.ttest_1samp(outcome[:, 0], outcome[:, 1], axis=None)
-        res2 = mstats.ttest_1samp(outcome[:, 0], outcome[:, 1], axis=None)
-        assert_allclose(res1, res2)
-
-        res1 = stats.ttest_1samp(outcome[:, :2], outcome[:, 2:], axis=0)
-        res2 = mstats.ttest_1samp(outcome[:, :2], outcome[:, 2:], axis=0)
-        assert_allclose(res1, res2, atol=1e-15)
-
-        # Check default is axis=0
-        res3 = mstats.ttest_1samp(outcome[:, :2], outcome[:, 2:])
-        assert_allclose(res2, res3)
-
```

Comment:
This test fails because the behavior didn't make sense before. Try the following in `main`:
```python3
import numpy as np
from scipy import stats

np.random.seed(1234567)
outcome = np.random.randn(20, 4) + [0, 0, 1, 2]

x = outcome[:, :2]
popmean = 0
popmeans = outcome[:, 2:]

res1 = stats.ttest_1samp(x, popmean, axis=None)  # res1.statistic.shape == ()
res2 = stats.ttest_1samp(x, popmeans, axis=None)  # res2.statistic.shape == (20, 2)
res3 = stats.ttest_1samp(x, popmeans, axis=0)  # res3.statistic.shape == (20, 2)
res4 = stats.ttest_1samp(x, popmeans, axis=1)  # Error: operands could not be broadcast together with shapes (20,) (20,2)
```
I think what was going on is that the mean of `x` is taken along `axis`, then `popmean` was subtracted. In the `axis=None` example, for instance, we take the mean of `x` with `axis=None` to get a scalar, and then we're effectively performing the test for this scalar against every entry in the array `popmean`. This doesn't seem intentional, or there would be no reason for the shape requirement:
![image](https://user-images.githubusercontent.com/6570539/184601308-f5d624a5-e65a-4dfc-8c52-dcb00698875e.png)

And if it were intentional, why does it fail with `axis=1`?

I think what was intended was for there to be one `popmean` per axis-slice of the sample `x`.

Reply:
> This test fails because the behavior didn't make sense before. Try the following in `main`:
> 
> ```python
> import numpy as np
> from scipy import stats
> 
> np.random.seed(1234567)
> outcome = np.random.randn(20, 4) + [0, 0, 1, 2]
> 
> x = outcome[:, :2]
> popmean = 0
> popmeans = outcome[:, 2:]
> 
> res1 = stats.ttest_1samp(x, popmean, axis=None)  # res1.statistic.shape == ()
> res2 = stats.ttest_1samp(x, popmeans, axis=None)  # res2.statistic.shape == (20, 2)
> res3 = stats.ttest_1samp(x, popmeans, axis=0)  # res3.statistic.shape == (20, 2)
> res4 = stats.ttest_1samp(x, popmeans, axis=1)  # Error: operands could not be broadcast together with shapes (20,) (20,2)
> ```
> 
> What was going on is that the mean of `x` was taken along `axis`, eliminating `axis` from the result, then `popmean` was subtracted. In the `axis=None` example, for instance, we take the mean of `x` with `axis=None` to get a scalar, and then we're effectively performing the test for this scalar sample mean against every entry in the array `popmean`. This doesn't seem intentional, or there would be no reason for the shape requirement as stated: ![image](https://user-images.githubusercontent.com/6570539/184601308-f5d624a5-e65a-4dfc-8c52-dcb00698875e.png)
> 
> `popmean` could be of _any_ shape when `axis=None`.
> 
> Also, was it intentional for this test to fail with `axis=1`?
> 
> I think what was intended was for there to be one `popmean` per axis-slice of the sample `x`. Let's follow standard broadcasting rules.

Yes, I too think this is the intended behavior.